### ISSUE TITLE
The authenticity of host 'x.x.x.x' can't be established

### ISSUE THREAD

**Issue by esinanturan** [trailofbits/algo#1583]
I am trying to setup algo vpn on my aws ec2 server on windows system but i get this error 

`UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: Warning: Permanently added 'x.x.x.x' (ECDSA) to the list of known hosts.\r\nubuntu@x.x.x.x: Permission denied (publickey).", "unreachable": true}`
I removed pem files in config directory and tried again as it said in troubleshooting page but still i get this error and setup cant be done successfully 

here is the setup process : 
[WARNING] Ansible is being run in a world writable directory (/mnt/c/Users/user/Desktop/algo-master), ignoring it as an ansible.cfg source. For more information see https://docs.ansible.com/ansible/devel/reference_appendices/config.html#cfg-in-world-writable-dir
 [WARNING]: Unable to parse /etc/ansible/hosts as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match
'all'

PLAY [localhost] *******************************************************************************************************
TASK [Gathering Facts] *************************************************************************************************ok: [localhost]

TASK [Ensure the requirements installed] *******************************************************************************ok: [localhost]

TASK [Set required ansible version as a fact] **************************************************************************ok: [localhost] => (item=ansible==2.7.12)
skipping: [localhost] => (item=netaddr)

TASK [Verify Ansible meets Algo VPN requirements.] *********************************************************************ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY [Ask user for the input] ******************************************************************************************
TASK [Gathering Facts] *************************************************************************************************ok: [localhost]

TASK [Cloud prompt] ****************************************************************************************************[Cloud prompt]
What provider would you like to use?
    1. DigitalOcean
    2. Amazon Lightsail
    3. Amazon EC2
    4. Vultr
    5. Microsoft Azure
    6. Google Compute Engine
    7. Scaleway
    8. OpenStack (DreamCompute optimised)
    9. CloudStack (Exoscale optimised)
    10. Install to existing Ubuntu 18.04 or 19.04 server (Advanced)

Enter the number of your desired provider
:
ok: [localhost]

TASK [Set facts based on the input] ************************************************************************************ok: [localhost]

TASK [VPN server name prompt] ******************************************************************************************[VPN server name prompt]
Name the vpn server
[algo]
:
ok: [localhost]

TASK [Cellular On Demand prompt] ***************************************************************************************[Cellular On Demand prompt]
Do you want macOS/iOS IPsec clients to enable "Connect On Demand" when connected to cellular networks?
[y/N]
:
ok: [localhost]

TASK [Wi-Fi On Demand prompt] ******************************************************************************************[Wi-Fi On Demand prompt]
Do you want macOS/iOS IPsec clients to enable "Connect On Demand" when connected to Wi-Fi?
[y/N]
:
ok: [localhost]

TASK [Trusted Wi-Fi networks prompt] ***********************************************************************************skipping: [localhost]

TASK [Retain the PKI prompt] *******************************************************************************************[Retain the PKI prompt]
Do you want to retain the keys (PKI)? (required to add users in the future, but less secure)
[y/N]
:
ok: [localhost]

TASK [DNS adblocking prompt] *******************************************************************************************[DNS adblocking prompt]
Do you want to enable DNS ad blocking on this VPN server?
[y/N]
:
ok: [localhost]

TASK [SSH tunneling prompt] ********************************************************************************************[SSH tunneling prompt]
Do you want each user to have their own account for SSH tunneling?
[y/N]
:
ok: [localhost]

TASK [Set facts based on the input] ************************************************************************************ok: [localhost]

PLAY [Provision the server] ********************************************************************************************
TASK [Gathering Facts] *************************************************************************************************ok: [localhost]

TASK [Display the invocation environment] ******************************************************************************
--> Please include the following block of text when reporting issues:

Algo running on: Ubuntu 18.04.2 LTS (Virtualized: wsl)
ZIP file created: 2019-09-06 11:56:24.000000000 +0300
Python 2.7.15+
Runtime variables:
    algo_provider "ec2"
    algo_ondemand_cellular "False"
    algo_ondemand_wifi "False"
    algo_ondemand_wifi_exclude "X251bGw="
    algo_dns_adblocking "True"
    algo_ssh_tunneling "False"
    wireguard_enabled "True"
    dns_encryption "True"
changed: [localhost -> localhost]

TASK [Install the requirements] ****************************************************************************************ok: [localhost -> localhost]

TASK [Generate the SSH private key] ************************************************************************************changed: [localhost]

TASK [Generate the SSH public key] *************************************************************************************changed: [localhost]

TASK [Include a provisioning role] *************************************************************************************
TASK [cloud-ec2 : Clean up the environment] ****************************************************************************skipping: [localhost]

TASK [cloud-ec2 : Install requirements] ********************************************************************************ok: [localhost]

TASK [cloud-ec2 : pause] ***********************************************************************************************[cloud-ec2 : pause]
Enter your aws_access_key (http://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html)
Note: Make sure to use an IAM user with an acceptable policy attached (see https://github.com/trailofbits/algo/blob/master/docs/deploy-from-ansible.md)
 (output is hidden):
ok: [localhost]

TASK [cloud-ec2 : pause] ***********************************************************************************************[cloud-ec2 : pause]
Enter your aws_secret_key (http://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html)
 (output is hidden):
ok: [localhost]

TASK [cloud-ec2 : set_fact] ********************************************************************************************ok: [localhost]

TASK [cloud-ec2 : Get regions] *****************************************************************************************ok: [localhost]

TASK [cloud-ec2 : Set facts about the regions] *************************************************************************ok: [localhost]

TASK [cloud-ec2 : Set the default region] ******************************************************************************ok: [localhost]

TASK [cloud-ec2 : pause] ***********************************************************************************************[cloud-ec2 : pause]
What region should the server be located in?
(https://docs.aws.amazon.com/general/latest/gr/rande.html#ec2_region)
    1. ap-northeast-1
    2. ap-northeast-2
    3. ap-south-1
    4. ap-southeast-1
    5. ap-southeast-2
    6. ca-central-1
    7. eu-central-1
    8. eu-north-1
    9. eu-west-1
    10. eu-west-2
    11. eu-west-3
    12. sa-east-1
    13. us-east-1
    14. us-east-2
    15. us-west-1
    16. us-west-2

Enter the number of your desired region
[13]
:
ok: [localhost]

TASK [cloud-ec2 : Set algo_region and stack_name facts] ****************************************************************ok: [localhost]

TASK [cloud-ec2 : Get existing available Elastic IPs] ******************************************************************skipping: [localhost]

TASK [cloud-ec2 : set_fact] ********************************************************************************************skipping: [localhost]

TASK [cloud-ec2 : pause] ***********************************************************************************************skipping: [localhost]

TASK [cloud-ec2 : set_fact] ********************************************************************************************skipping: [localhost]

TASK [cloud-ec2 : Locate official AMI for region] **********************************************************************ok: [localhost]

TASK [cloud-ec2 : Set the ami id as a fact] ****************************************************************************ok: [localhost]

TASK [cloud-ec2 : Deploy the template] *********************************************************************************
changed: [localhost]

TASK [cloud-ec2 : set_fact] ********************************************************************************************
ok: [localhost]

TASK [Set subjectAltName as afact] *************************************************************************************
ok: [localhost]

TASK [Add the server to an inventory group] ****************************************************************************
changed: [localhost]

TASK [Additional variables for the server] *****************************************************************************
changed: [localhost]

TASK [Wait until SSH becomes ready...] *********************************************************************************
ok: [localhost]

TASK [MacOS | set OS specific facts] ***********************************************************************************
skipping: [localhost]

TASK [MacOS | mount a ram disk] ****************************************************************************************
skipping: [localhost]

TASK [Linux | set OS specific facts] ***********************************************************************************
ok: [localhost]

TASK [Set config paths as facts] ***************************************************************************************
ok: [localhost]

TASK [Update config paths] *********************************************************************************************
changed: [localhost]

TASK [debug] ***********************************************************************************************************
ok: [localhost] => {
    "IP_subject_alt_name": "3.123.215.33"
}

TASK [A short pause, in order to be sure the instance is ready] ********************************************************
Pausing for 20 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)
ok: [localhost]

PLAY [Configure the server and install required software] **************************************************************

TASK [common : Check the system] ***************************************************************************************
The authenticity of host '3.123.215.33 (3.123.215.33)' can't be established.
ECDSA key fingerprint is SHA256:IXiIObQixtMrT3Z4z9/QiSlwiO4LyqkeN0zs82+QeBk.
Are you sure you want to continue connecting (yes/no)? yes
fatal: [3.123.215.33]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: Warning: Permanently added '3.123.215.33' (ECDSA) to the list of known hosts.\r\nubuntu@3.123.215.33: Permission denied (publickey).", "unreachable": true}
        to retry, use: --limit @/mnt/c/Users/user/Desktop/algo-master/main.retry

PLAY RECAP *************************************************************************************************************
3.123.215.33               : ok=0    changed=0    unreachable=1    failed=0
localhost                  : ok=41   changed=7    unreachable=0    failed=0

**Comment by esinanturan** [comment]
I have cloned algo repo into home directory and tried but still getting same error

**Comment by jackivanov** [comment]
@esinanturan Did you check out #1584?

**Comment by esinanturan** [comment]
Yes I tried but I still get same error in windows

**Comment by davidemyers** [comment]
@esinanturan  How about if you run the following command from the `algo` directory right after you clone the source into your home directory:

```
chmod o-w .
```

**Comment by wizonesolutions** [comment]
Getting this too. My pubkey is in the agent. Does Algo manually look for `id_rsa` hardcoded or something? I can SSH into the server fine from the same shell (WSL 2 Ubuntu) where I'm running Algo.

**Comment by davidemyers** [comment]
@wizonesolutions I use an encrypted `algo.pem` and `ssh-agent` for my Algo deployments and have no problems. But I don't have Windows or use WSL.

Can you check the permissions of your Algo directory and make sure it's not world-writable? [According to Ansible](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#avoiding-security-risks-with-ansible-cfg-in-the-current-directory) that could break things.

**Comment by wizonesolutions** [comment]
I will check that when back in Windows. I did get the VPN deployed (copying the key to `id_rsa` isn't a tough workaround). I'm not sure why that would prevent it from seeing the keys loaded in the agent via `SSH_AUTH_SOCK`. I'm also on WSL 2, which is basically a glorified Linux VM...so the environment should be very Linuxy. I probably missed something and need to look more closely at auth.log. But I got things up and running, which was the main thing. Had to try twice because it turns out I needed to tweak `reduce_mtu`.
